app/policies/application_policy.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class ApplicationPolicy
+  RESOURCE_ACTIONS = %i[index? create? new? update? edit? destroy?].freeze
+  attr_reader :user, :record
+
+  def initialize(user, record)
+    @user = user
+    @record = record
+    require_user! if user_required?
+  end
+
+  def method_missing(_method_name, *_args)
+    false
+  end
+
+  def respond_to_missing?(method_name, include_private = false)
+    RESOURCE_ACTIONS.include?(method_name) || super
+  end
```

Comment:
That's an interesting part; I am not quite sure I get the motivation correctly. I assume you wanted to provide a "fallback", in case something goes wrong. Having this fallback to be false makes sense, absolutely. 

Then, you are also overwriting the `respond_to_missing?`, which seems to be good practice when overwriting `method_missing` (I just learned that -- thanks!). Nevertheless, I now feel there is some mismatch between`.respond_to? :foo`, `.method :foo`, and `.foo`.
[Read more](https://blog.marc-andre.ca/2010/11/15/methodmissing-politely/)

```rb
[13] pry(main)> ap = ApplicationPolicy.new("bar", "foobar")
[13] pry(main)> ap.foo
=> false
[14] pry(main)> ap.method :foo
NameError: undefined method `foo' for class `ApplicationPolicy'
from (pry):14:in `method'
[15] pry(main)> ap.respond_to? :foo
=> false
```

Reply:
We need every policy to not throw a MethodNotFound for the standard actions like index?, show? etc. because we use those in views/application/_breadcrumbs_and_title.html.slim. 
We could just return false for those in the ApplicationPolicy but then every policy would have those public instance methods even when a controller does not have an action like that. This causes a problem in the policy specs because then we cannot write 
```
context 'when current_user is admin' do
      let(:current_user) { create(:admin) }

      it { is_expected.to permit_all_actions }
end
```
This would fail for the users controller because there is no index route for users which means even admins cannot access this. We would have to write something like the following instead:
```
context 'when current_user is admin' do
      let(:current_user) { create(:admin) }

      it { is_expected.to forbid_only_actions(%i[index ...]) }
end
```
This makes the specs harder to read which I wanted to avoid. The solution I came to was this little meta programming stunt. Because the index?, show?, ... actions are technically not defined the pundit_matcher permit_all_actions is happy and the breadcrumbs partial is also happy because it gets a return value for the standard actions. Maybe you have another idea, though?

Reply:
Ah, I see. Overall, the solution you've come up with is quite clever, and I like it :+1:. It fits great for both the view (breadcrumbs_and_title) as well as the specs. The only thing might be some more complicated "debugging" if something is not working or someone forgot to specify the respective method on their own. This could be something optional to include as a debug (or trace) log output. Otherwise, I think that's fine.

Reply:
I added debug output like you suggested.